compiler/rustc_abi/src/lib.rs

```diff
@@ -967,6 +1006,28 @@ impl WrappingRange {
         }
     }
 
+    /// Returns `true` if `vmin..=vmax` is contained in `self`.
+    #[inline(always)]
+    pub fn contains_range(&self, vmin: u128, vmax: u128) -> bool {
+        assert!(vmin <= vmax);
+        if self.start <= self.end {
+            self.start <= vmin && vmax <= self.end
+        } else {
+            self.start <= vmin || vmax <= self.end || self.start == self.end + 1
```

Comment:
The `self.start == self.end + 1` part here seems unnecessary? If `self.start <= vmin || vmax <= self.end` is false then we have `end > vmax >= vmin > start`, so end is at least 2 above start, and `start == end + 1` cannot possibly be true.

Reply:
```suggestion
            // `self.start == self.end + 1` is needed to cover this case:
            // vmin ... start, end ... vmax. In this special case there is no gap
            // between `start` and `end` so we must return true.
            self.start <= vmin || vmax <= self.end || self.start == self.end + 1
```